pilot/pkg/security/authz/builder/extauthz.go

```diff
@@ -161,6 +162,18 @@ func buildExtAuthzHTTP(in *plugin.InputParams, config *meshconfig.MeshConfig_Ext
 			errs = multierror.Append(errs, fmt.Errorf("pathPrefix must begin with `/`, found: %q", config.PathPrefix))
 		}
 	}
+	checkWildcard := func(field string, values []string) {
+		for _, val := range values {
+			if val == "*" {
+				errs = multierror.Append(errs, fmt.Errorf("a single wildcard (\"*\") is not supported, change it to either prefix or suffix match: %s", field))
```

Comment:
it is allowed but here we want to make the syntax consistent with the authz policy, in authz, a single `*` means presence match where does not apply to ext-authz filter which I think it's then better rejected early in validation step.

Reply:
Should this be in the mesh config validation? I think that will fail at install time instead of here. we can have it in both if we need to

Reply:
I think we actually don't reject the config in mesh config validation, it only outputs the error message in istiod.

We also don't want to reject because a problematic config may not be used by any actual authorization policy and we don't want it to block the process, that's why we actually changed this in 1.9.1 from the initial implementation.

We will generate a default deny-all config config if any errors found later when actually processing the authz policy, and also output the error message in istiod.

Reply:
maybe we should do it in both places then? We definitely block install if its invalid, which I think is a good time to catch it?

Reply:
that's actually what we did in the initial implementation. There is feedback about the inconvenience that an invalid config could block the install even if it is not actively used. For example, the ext-authz service is not installed/known to istio yet or experimenting some test configs that may not work already.

For all these cases, I think a better user experience is to block when the invalid config is actually used by authorization policies (current behavior), instead of at the moment of defining the config (but not used yet).

Reply:
I am not sure why we would accept invalid config. Its one thing if its temporarily invalid (reference service that doesn't exist), but in this case its wrong 100% of the time. I cannot see a circumstance where a user explicitly sets a field and would prefer for it to fail later in logs rather than immediately.

Reply:
We are not really accepting the invalid config, we are just not rejecting it at the admission moment but will later reject it when it's being used. This is because the majority invalid configs are actually temporarily invalid (e.g. service not ready) and should become ready at a later point.

If we want to reject mesh config in the admission time, we need to only reject the invalid configs that are always invalid and not those temporarily invalid configs. In addition to this "*" check introduced in this PR, there are probably some common checks for the extension provider, like the config name must be unique and not empty, I will look into this in a follow PR.

Reply:
Yeah I definitely agree about the temporarily invalid ones shouldn't be in the validation. Validation should be stateless and look at only the single resource in isolation